src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -57,11 +67,31 @@ public Resource(IEnumerable<KeyValuePair<string, object>> attributes)
         /// </summary>
         public static Resource Empty { get; } = new Resource(Enumerable.Empty<KeyValuePair<string, object>>());
 
+        /// <summary>
+        /// Gets the dafault Resource with telemetry sdk attributes set.
+        /// </summary>
+        public static Resource Default { get; } = new Resource(new List<KeyValuePair<string, object>>
+            {
+                new KeyValuePair<string, object>(TelemetrySdkNameKey, "opentelemetry"),
+                new KeyValuePair<string, object>(TelemetrySdkLanguageKey, "dotnet"),
+                new KeyValuePair<string, object>(TelemetrySdkVersionKey, Version.ToString()),
+            });
+
         /// <summary>
         /// Gets the collection of key-value pairs describing the resource.
         /// </summary>
         public IEnumerable<KeyValuePair<string, object>> Attributes { get; }
 
+        /// <summary>
+        /// Creates a new <see cref="Resource"/> with user provided attributes, telemetry sdk attributes, and attributes from OTEL_RESOURCE_ATTRIBUTES environment variable.
+        /// </summary>
+        /// <param name="attributes">An <see cref="IDictionary{String, Object}"/> of attributes that describe the resource.</param>
+        /// <returns><see cref="Resource"/>.</returns>
+        public static Resource Create(IEnumerable<KeyValuePair<string, object>> attributes)
```

Comment:
The name is bit misleading.. as it doesn't tell that this method will merge with default and also reads from ENV..

Reply:
The API doc for the method describe the operations done by the `Create` method. I wanted to keep the name simple and inline with the [Python](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py#L109) and [Java](https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L130) counterparts.
We can call it `CreateWithDefault` but that would need to change if the method operations are modified in the future. Thoughts?